### PR TITLE
Allow to specify the path to config file

### LINKED ISSUE #78: Allow to specify the path to config file
Sometimes you have multiple solidity projects that want to share the same solhint config. Now the only option is to copy the config file. If we allow to specify the config file location instead of hardcoding it we will be able to put that file in one place and reference it.

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #78

### CODE REVIEW

File: solhint.js [hunk 1]
```diff
@@ -16,6 +19,7 @@ function init() {
       '-w, --max-warnings [maxWarningsNumber]',
       'number of warnings to trigger nonzero exit code'
     )
+    .option('-c, --config-path [file_name]', 'file to use as your .solhint.json')
```
Comment (fvictorio) [inline solhint.js]: Let's use `--config` here, like eslint does.

File: solhint.js [hunk 2]
```diff
@@ -5,9 +5,12 @@ const _ = require('lodash')
 const fs = require('fs')
 const process = require('process')
 const linter = require('./lib/index')
+const packageJson = require('./package.json')
+const { FileNotExistsError } = require('./lib/common/errors')
 
 function init() {
-  program.version('1.1.10')
+  const version = packageJson.version || '1.1.10'
```
Comment (fvictorio) [inline solhint.js]: I think we can remove the `1.1.10` fallback here.

File: solhint.js [hunk 3]
```diff
@@ -128,13 +132,30 @@ const readConfig = _.memoize(() => {
...
+    }
+
     config = JSON.parse(configStr)
   } catch (e) {
     if (e instanceof SyntaxError) {
       console.log('ERROR: Configuration file [.solhint.json] is not a valid JSON!\n')
       process.exit(0)
     }
+    if (e instanceof FileNotExistsError) {
+      console.log('ERROR: Configuration file [' + program.configPath + '] doesnt exist!\n')
```
Comment (fvictorio) [inline solhint.js]: `doesn't`

File: solhint.js [hunk 4]
```diff
@@ -128,13 +132,30 @@ const readConfig = _.memoize(() => {
...
+      process.exit(0)
```
Comment (fvictorio) [inline solhint.js]: This shouldn't exit with a success code.

File: solhint.js [hunk 5]
```diff
@@ -128,13 +132,30 @@ const readConfig = _.memoize(() => {
   let config = {}
 
   try {
-    const configStr = fs.readFileSync('.solhint.json').toString()
+    let configStr
```
Comment (fvictorio) [inline solhint.js]: I think we should do:

```js
const configStr = program.configPath || '.solhint.json'
```

and then adapt the following lines accordingly.

File: lib/common/errors.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class FileNotExistsError extends Error {
+  constructor(message, status) {
+    super(message)
+
+    this.name = this.constructor.name
+
+    Error.captureStackTrace(this, this.constructor)
+
+    this.status = status || 500
```
Comment (fvictorio) [inline lib/common/errors.js]: This seems copied from an error for HTTP. Can we simplify it?

File: solhint.js (hunk 1 above)
Comment (mariano-aguero) [reply solhint.js]: Solved

File: solhint.js (hunk 2 above)
Comment (mariano-aguero) [reply solhint.js]: Solved

File: solhint.js (hunk 3 above)
Comment (mariano-aguero) [reply solhint.js]: Solved!

File: solhint.js (hunk 5 above)
Comment (mariano-aguero) [reply solhint.js]: Solved!

File: lib/common/errors.js (hunk 6 above)
Comment (mariano-aguero) [reply lib/common/errors.js]: Solved!

File: solhint.js (hunk 4 above)
Comment (mariano-aguero) [reply solhint.js]: Solved!

File: solhint.js [hunk 7]
```diff
@@ -128,12 +132,23 @@ const readConfig = _.memoize(() => {
...
     config = JSON.parse(configStr)
   } catch (e) {
     if (e instanceof SyntaxError) {
       console.log('ERROR: Configuration file [.solhint.json] is not a valid JSON!\n')
-      process.exit(0)
+      process.exit(1)
+    }
+    if (e instanceof FileNotExistsError) {
+      console.log(`ERROR: Configuration file [${program.config}] doesn't exist!\n`)
+      process.exit(1)
```
Comment (pablofullana) [inline solhint.js]: Can't this two `process.exit(1)` lines be simplified in just one after both `if` statements?

Comment (mariano-aguero) [reply solhint.js]: Done

**Comment by mariano-aguero** [comment]
Closes #84

**Pull request merged**